src/main/java/org/jenkinsci/remoting/protocol/IOHub.java

```diff
@@ -489,6 +515,53 @@ public final void run() {
             // ignore, happens routinely
         } finally {
             selectorThread.setName(oldName);
+            ioHubRunning = false;
```

Comment:
at this point `selector.isOpen()` should return `false` - so can you get rid of this?

Reply:
Only if it is aborted properly. I see hanging of the selector close() operation in some unit tests